src/main/java/frc/robot/subsystems/elevator/factories/RealElevatorConstants.java

```diff
@@ -0,0 +1,100 @@
+
+import com.revrobotics.CANSparkBase;
+import com.revrobotics.SparkLimitSwitch;
+import edu.wpi.first.math.controller.ElevatorFeedforward;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.wpilibj2.command.sysid.SysIdRoutine;
+import frc.robot.constants.IDs;
+import edu.wpi.first.math.filter.Debouncer;
+import frc.robot.hardware.digitalinput.IDigitalInput;
+import frc.robot.hardware.digitalinput.supplied.SuppliedDigitalInput;
+import frc.robot.hardware.motor.ControllableMotor;
+import frc.robot.hardware.motor.sparkmax.BrushlessSparkMAXMotor;
+import frc.robot.hardware.motor.sparkmax.SparkMaxWrapper;
+import frc.robot.hardware.request.cansparkmax.SparkMaxAngleRequest;
+import frc.robot.hardware.signal.supplied.SuppliedAngleSignal;
+import frc.robot.hardware.signal.supplied.SuppliedDoubleSignal;
+import frc.robot.subsystems.elevator.ElevatorConstants;
+import frc.robot.subsystems.elevator.ElevatorMotorStuff;
+import frc.robot.subsystems.elevator.ElevatorStuff;
+import frc.utils.AngleUnit;
+
+import java.util.function.BooleanSupplier;
+import java.util.function.Supplier;
+
+public class RealElevatorConstants {
+
+	private static final double DEBOUNCE_TIME_SECONDS = 0.05;
+
+	private static final Debouncer.DebounceType DEBOUNCE_TYPE = Debouncer.DebounceType.kBoth;
+
+	private static final CANSparkBase.SoftLimitDirection SOFT_LIMIT_DIRECTION = CANSparkBase.SoftLimitDirection.kReverse;
+
+	private static final int ELEVATOR_PID_SLOT = 0;
+
+	private static final Rotation2d REVERSE_SOFT_LIMIT_VALUE = Rotation2d.fromRotations(0);
+
+	private static final double MOTOR_ROTATIONS_TO_METERS_CONVERTION_RATIO = 1;
+
+	private static final SparkLimitSwitch.Type REVERSE_LIMIT_SWITCH_TYPE = SparkLimitSwitch.Type.kNormallyOpen;
+
+	private static final ElevatorFeedforward FEEDFORWARD_CALCULATOR = new ElevatorFeedforward(0, 0, 0, 0);
+
+	public static double feedforwardCalculation(Rotation2d velocity) {
+		return RealElevatorConstants.FEEDFORWARD_CALCULATOR.calculate(velocity.getRadians());
+	}
+
+	public static void configureMotor(SparkMaxWrapper sparkMaxWrapper) {
+		sparkMaxWrapper.setSoftLimit(SOFT_LIMIT_DIRECTION, (float) REVERSE_SOFT_LIMIT_VALUE.getRotations());
+		sparkMaxWrapper.enableSoftLimit(SOFT_LIMIT_DIRECTION, true);
+		sparkMaxWrapper.getEncoder().setPositionConversionFactor(ElevatorConstants.GEAR_RATIO);
+		sparkMaxWrapper.getEncoder().setVelocityConversionFactor(ElevatorConstants.GEAR_RATIO);
+		sparkMaxWrapper.getPIDController().setP(1);
+		sparkMaxWrapper.getPIDController().setI(0);
+		sparkMaxWrapper.getPIDController().setD(0);
+	}
+
+	public static ElevatorMotorStuff generateMotorStuff(String logPath, String motorName, SparkMaxWrapper sparkMaxWrapper) {
+		configureMotor(sparkMaxWrapper);
+
+		ControllableMotor motor = new BrushlessSparkMAXMotor(logPath, sparkMaxWrapper, new SysIdRoutine.Config());
+
+		Supplier<Double> motorPosition = () -> sparkMaxWrapper.getEncoder().getPosition();
```

Comment:
```suggestion
		Supplier<Double> motorPosition = sparkMaxWrapper.getEncoder() :: getPosition();
```

Reply:
u cant do this only if it was on the sparkMaxWrapper

Reply:
![image](https://github.com/user-attachments/assets/278e73a0-06a4-44a6-bff9-81e2aca6c56c)